openedx/core/djangoapps/schedules/signals.py

```diff
@@ -28,58 +28,49 @@
-    # either the enrollment date or the course's start date.
-    content_availability_date = max(enrollment.created, enrollment.course_overview.start)
-
-    upgrade_deadline = _calculate_upgrade_deadline(enrollment.course_id, content_availability_date)
-
-    if course_has_highlights(enrollment.course_id):
-        experience_type = ScheduleExperience.EXPERIENCES.course_updates
-    else:
-        experience_type = ScheduleExperience.EXPERIENCES.default
-
-    if _should_randomly_suppress_schedule_creation(
-        schedule_config,
-        enrollment,
-        upgrade_deadline,
-        experience_type,
-        content_availability_date,
-    ):
-        return
-
-    schedule = Schedule.objects.create(
-        enrollment=enrollment,
-        start=content_availability_date,
-        upgrade_deadline=upgrade_deadline
-    )
+    try:
+        enrollment = kwargs['instance']
+        current_site = get_current_site()
+        if not _should_create_schedule(kwargs['created'], current_site, enrollment):
+            return
+
+        schedule_config = ScheduleConfig.current(current_site)
+
+        # This represents the first date at which the learner can access the content. This will be the latter of
+        # either the enrollment date or the course's start date.
+        content_availability_date = max(enrollment.created, enrollment.course_overview.start)
+        upgrade_deadline = _calculate_upgrade_deadline(enrollment.course_id, content_availability_date)
+        experience_type = _get_experience_type(enrollment)
+
+        if _should_randomly_suppress_schedule_creation(
+                schedule_config,
+                enrollment,
+                upgrade_deadline,
+                experience_type,
+                content_availability_date,
+        ):
+            return
+
+        schedule = Schedule.objects.create(
+            enrollment=enrollment,
+            start=content_availability_date,
+            upgrade_deadline=upgrade_deadline
+        )
 
-    ScheduleExperience(schedule=schedule, experience_type=experience_type).save()
+        ScheduleExperience(schedule=schedule, experience_type=experience_type).save()
 
-    log.debug('Schedules: created a new schedule starting at %s with an upgrade deadline of %s and experience type: %s',
-              content_availability_date, upgrade_deadline, ScheduleExperience.EXPERIENCES[experience_type])
+        log.debug(
+            'Schedules: created a new schedule starting at %s with an upgrade deadline of %s and experience type: %s',
+            content_availability_date, upgrade_deadline, ScheduleExperience.EXPERIENCES[experience_type]
+        )
+    except Exception:  # pylint: disable=broad-except
+        # We do not want to block the creation of a CourseEnrollment because of an error in creating a Schedule.
+        # No Schedule is acceptable, but no CourseEnrollment is not.
+        enrollment = kwargs.get('instance')
```

Comment:
Can you move this line out of the try-except block so it can be computed once and used within both places?